rt-crowd-forecast/update-samples.R

```diff
@@ -0,0 +1,205 @@
+                                              sheet = "ids"))
+  
+# load forecasts 
+forecasts <- try_and_wait(googlesheets4::read_sheet(ss = spread_sheet))
+
+
+delete_data <- FALSE
+if (delete_data) {
+  # add forecasts to backup sheet
+  try_and_wait(googlesheets4::sheet_append(ss = spread_sheet, 
+                                           sheet = "oldforecasts", 
+                                           data = forecasts))
+  
+  # delete data from sheet
+  cols <- data.frame(matrix(ncol = ncol(forecasts), nrow = 0))
+  names(cols) <- names(forecasts)
+  try_and_wait(googlesheets4::write_sheet(data = cols,
+                                          ss = spread_sheet,
+                                          sheet = "predictions"))
+}
+
+# obtain raw and filtered forecasts, save raw forecasts-------------------------
+raw_forecasts <- forecasts %>%
+  dplyr::mutate(forecast_date = as.Date(forecast_date), 
+                submission_date = as.Date(submission_date))
+
+# use only the latest forecast from a given forecaster
+filtered_forecasts <- raw_forecasts %>%
+  # interesting question whether or not to include foracast_type here. 
+  # if someone reconnecs and then accidentally resubmits under a different
+  # condition should that be removed or not? 
+  dplyr::group_by(forecaster_id, region, target_type) %>%
+  dplyr::filter(forecast_time == max(forecast_time)) %>%
+  dplyr::ungroup() 
+
+# replace forecast duration with exact data about forecast date and time
+# define function to do this for raw and filtered forecasts
+replace_date_and_time <- function(forecasts) {
+  forecast_times <- forecasts %>%
+    dplyr::group_by(forecaster_id, region) %>%
+    dplyr::summarise(forecast_time = unique(forecast_time)) %>%
+    dplyr::ungroup() %>%
+    dplyr::arrange(forecaster_id, forecast_time) %>%
+    dplyr::group_by(forecaster_id) %>%
+    dplyr::mutate(forecast_duration = c(NA, diff(forecast_time))) %>%
+    dplyr::ungroup()
+  
+  forecasts <- dplyr::inner_join(forecasts, forecast_times, 
+                                 by = c("forecaster_id", "region", "forecast_time")) %>%
+    dplyr::mutate(forecast_week = lubridate::epiweek(forecast_date), 
+                  target_end_date = as.Date(target_end_date)) %>%
+    dplyr::select(-forecast_time)
+  
+  return(forecasts)
+}
+
+# replace time with duration and date with epiweek
+raw_forecasts <- replace_date_and_time(raw_forecasts)
+filtered_forecasts <- replace_date_and_time(filtered_forecasts)
+
+# write raw forecasts
+data.table::fwrite(raw_forecasts %>%
+                     dplyr::select(-board_name),
+                   paste0("rt-crowd-forecast/raw-forecast-data/", 
+                          submission_date, "-raw-forecasts.csv"))
+
+# draw samples from the distributions ------------------------------------------
+n_people <- filtered_forecasts$forecaster_id %>%
+  unique() %>%
+  length()
+
+draw_samples <- function(distribution, 
```

Comment:
is this the same as in the other script?

Reply:
similar. in the other script it deals with quantiles, not samples, so not sure it's worth the effort of writing one function for both